lib/keychain.js

```diff
@@ -76,6 +84,18 @@ class Keychain {
     ])
   }
 
+  buildTransaction(transaction, service, index) {
+    const keypair = this.deriveKeypair(service, index)
+    const address = this.deriveAddress(service, index + 1)
+
+    transaction.setAddress(address)
+
+    const payloadForPreviousSignature = transaction.previousSignaturePayload()
+    const previousSignature = sign(payloadForPreviousSignature, keypair.privateKey)
+    
+    transaction.setPreviousSignatureAndPreviousPublicKey(previousSignature, keypair.publicKey)
```

Comment:
WIll be great to return the transaction builder to avoid any confusion (FP style)